dimscord/restapi.nim

```diff
@@ -347,65 +347,41 @@ proc getChannelPins*(api: RestApi, channel_id: string): Future[seq[Message]] {.a
         for pin in pins.elems:
             result.add(newMessage(pin))
 
+macro loadOpt(obj: typed, lits: varargs[untyped]): untyped =
+    # Best demonstrated through example.
+    # loadOpt(payload, name, position, topic) expands to:
+    # if name.isSome:
+    #     payload["name"] = %get(name)
+    # if position.isSome:
+    #     payload["position"] = %get(position)
+    # if topic.isSome:
+    #     payload["topic"] = %get(topic)
+    result = newStmtList()
+    for lit in lits:
+        let fieldName = lit.strVal
+        result.add quote do:
+            if `lit`.isSome:
+                `obj`[`fieldName`] = %get(`lit`)
+
 proc editGuildChannel*(api: RestApi, channel_id: string; name: Option[string] = none(string); position: Option[int] = none(int);
             topic: Option[string] = none(string); nsfw: Option[bool] = none(bool); rate_limit_per_user: Option[int] = none(int); bitrate: Option[int] = none(int);
             user_limit: Option[int] = none(int); permission_overwrites: Option[seq[Overwrite]] = none(seq[Overwrite]); parent_id: Option[string] = none(string); reason: string = ""): Future[GuildChannel] {.async.} =
     let h = if reason != "": newHttpHeaders({"X-Audit-Log-Reason": reason}) else: nil
-    let payload = newJObject()
+    var payload = %*{} # Don't judge me.
 
-    if name.isSome:
-        payload["name"] = %get(name)
-    if position.isSome:
-        payload["position"] = %get(position)
-    if topic.isSome:
-        payload["topic"] = %get(topic)
-    if nsfw.isSome:
-        payload["nsfw"] = %get(nsfw)
-    if rate_limit_per_user.isSome:
-        payload["rate_limit_per_user"] = %get(rate_limit_per_user)
-    if bitrate.isSome:
-        payload["bitrate"] = %get(bitrate)
-    if user_limit.isSome:
-        payload["user_limit"] = %get(user_limit)
-    if permission_overwrites.isSome:
-        var perms: seq[JsonNode] = @[]
-
-        for ow in get(permission_overwrites):
-            perms.add(%ow)
-        payload["permission_overwrites"] = %perms

```

Comment:
From my testing, it seems that `%` serializes the elements of a `seq` recursively, so this is not necessary. The macro replacement is `payload["permission_overwrites"] = %get(permission_overwrites)`.